calico/openstack/test/test_plugin.py

```diff
@@ -0,0 +1,458 @@
+                         if socket.bound_address == "tcp://*:9903"}
+        assert len(bound_sockets) == 1
+        self.acl_get_socket = bound_sockets.pop()
+        print "ACL GET socket is %s" % self.acl_get_socket
+
+        #*********************************************************************#
+        #* Check that there's a socket bound to port 9904, and get it.       *#
+        #*********************************************************************#
+        bound_sockets = {socket for socket in self.sockets
+                         if socket.bound_address == "tcp://*:9904"}
+        assert len(bound_sockets) == 1
+        self.acl_pub_socket = bound_sockets.pop()
+        print "ACL PUB socket is %s" % self.acl_pub_socket
+
+        #*********************************************************************#
+        #* Hook the Neutron database.                                        *#
+        #*********************************************************************#
+        self.db = mech_calico.manager.NeutronManager.get_plugin()
+        self.db_context = mech_calico.ctx.get_admin_context()
+        self.db.get_ports.return_value = []
+
+        #*********************************************************************#
+        #* Send a RESYNCSTATE.                                               *#
+        #*********************************************************************#
+        resync = {'type': 'RESYNCSTATE',
+                  'resync_id': 'resync#1',
+                  'issued': time.time() * 1000,
+                  'hostname': 'felix-host-1'}
+        self.felix_router_socket.rcv_queue.put_nowait(
+            ['felix-1',
+             '',
+             json.dumps(resync).encode('utf-8')])
+        self.real_eventlet_sleep(REAL_EVENTLET_SLEEP_TIME)
+
+        #*********************************************************************#
+        #* Check DB got create_or_update_agent call.                         *#
+        #*********************************************************************#
+        self.db.create_or_update_agent.assert_called_once_with(
+            self.db_context,
+            {'agent_type': mech_calico.AGENT_TYPE_FELIX,
+             'binary': '',
+             'host': 'felix-host-1',
+             'topic': mech_calico.constants.L2_AGENT_TOPIC,
+             'start_flag': True})
+
+        #*********************************************************************#
+        #* Check RESYNCSTATE response was sent.                              *#
+        #*********************************************************************#
+        self.felix_router_socket.send_multipart.assert_called_once_with(
+            ['felix-1',
+             '',
+             json.dumps({'type': 'RESYNCSTATE',
+                         'endpoint_count': 0,
+                         'rc': 'SUCCESS',
+                         'message': 'Здра́вствуйте!'}).encode('utf-8')])
+        self.felix_router_socket.send_multipart.reset_mock()
```

Comment:
So long as 'more data' is 'even one test failure caused by out-of-order dictionaries', then yes. =)